### ISSUE TITLE
Server error probably caused by exceeded a character limit

### ISSUE THREAD

**Issue by lowwintersun** [carboncoop/home-energy-health-check#21]
Rae has an issue where she had same error as shown below when hitting the save and quit button, when she went back to the form the comfort and health info was still present but all the info in the measures fields had disappeared. 

![pec issue image003](https://user-images.githubusercontent.com/7352302/32609456-bae3911a-c557-11e7-95e2-c63c36b29a34.jpg)

**Comment by AppSynergy** [comment]
What url is this screenshot taken from?

The red bar is an error alert - but it should contain the error message. It's possible the error doesn't have one. If an error is returned, the assessment is not saved.

Possible duplicate #21

**Comment by lowwintersun** [comment]
I will seek more information on this as the assessor who reported #21 has reported this issue second hand.

**Comment by AppSynergy** [comment]
Looks like an error of some sort when sending the http request.

I've left some [console warnings](https://github.com/carboncoop/home-energy-health-check/blob/develop/resources/assets/js/mixins/form.js#L49) to trigger when it fails.

If anyone manages to get an empty red alert like this, can you look in the javascript console? There should be a bit of info there.

**Comment by lowwintersun** [comment]
Will do - I'm having trouble replicating the issue. Maybe it's browser or operating system specific? They're all using Microsoft Surfaces (which are Council issue and they've complained about a lot!).

**Comment by lowwintersun** [comment]
No report of repeat.

**Comment by lowwintersun** [comment]
This is really perplexing me but we have a console warning now - see below 'Server Error'

This occurs with just one assessor when they try to 'save and quit' or 'submit' she gets the server error below and the information in the assessment won’t pull through and completely disappears.

She actually has the populated assessment saved locally but says when she tries to submit it to the server she can't (I don't know if she gets the same server warning).

I don't know if this has happened with other assessors or with the same assessor on different assessments. 

Any thoughts

![image](https://user-images.githubusercontent.com/7352302/32851944-629c608c-ca2e-11e7-8507-0431a41a160f.png)

**Comment by AppSynergy** [comment]
Again, it'd be useful to know what url is this screenshot taken from?

Oh, and is it possible to expand and grab the console warning that accompanies it to?

**Comment by lowwintersun** [comment]
I think it's here: http://seem-energy-check.org.uk/submit/48/edit#/submit

They are using Microsoft Surface computers - could it be that they are using an outdated version of IE? I believe JS support on IE can be dodgy.

I will ask them to copy and the paste the console data but maybe we should think about using something like Teamviewer as their technical competence is low. Either you or I could do it with the assessor in question?

**Comment by AppSynergy** [comment]
> JS support on IE can be dodgy.

Ooh yes, it can. I thought Microsoft Surfaces are modern things though.. aren't they all running Edge?

Either way, it's all packed with webpack, babel et al straight out-of-the-box so should be back-compatible as far as the authors intended...

They're also clearly able to run the majority of the front-end application, with just the axios http request failing maybe?

Can we confirm the OS and browser? You may be onto something...

**Comment by lowwintersun** [comment]
Cool, I've asked the question. When I was down there they were all grumbling about the Surfaces, I think the council had recently stopped issuing them shortly after they got them all!

**Comment by AppSynergy** [comment]
Other questions worth asking:

* Is the issue intermittent, or firmly repeatable?
* Does the affected user have issues clicking "Save and Quit" on **all** Assessments, or just this one?
* Can they create new Assessments, or do they get the same error?

**Comment by lowwintersun** [comment]
A coworker in Plymouth has repeated the error ok their PC using the original assement text. They speculate some whether some sections have a character count limit and whether server error appears when this is surpassed? I’m out of the office so can’t replicate to confirm for now but might be on to something?

**Comment by AppSynergy** [comment]
I like the sound of that one - character limits are set by the database and are easily re-configured in theory.

Can we get any of the following: ?

* how many characters did he enter?
* which inputs did he enter them in?
* what should we consider the max limits to be? (per column, if necessary)

**Comment by AppSynergy** [comment]
I've got a feeling it might be `comfort_favourite_room` and/or `comfort_least_loved_room` causing the problem.

Both are `varchar(255)` in the database, as they looked like they'd just be a few words, but I've assigned them multi-line textareas for some reason, and users are taking the opportunity to write more.

Even if that's somehow not the problem here it would have cropped up soooner or later.

I've released a patch (0e943786b43a904) to https://seem-energy-check.org.uk/, that changes these two fields to arbitrarily long text.

Can we confirm this was the (only) problem?
